### PR TITLE
Filters: configurable and arbitrary number of parameters

### PR DESCRIPTION
**Pull request by wagner-certat**
See NEWS.md:

 * Events: Subqueries can be extended by the configuration, for example:
   ```json
   "subqueries": {
     "all_ips": {
       "sql": "(\"source.ip\" = %s OR \"source.local_ip\" = %s OR \"destination.ip\" = %s OR \"destination.local_ip\" = %s)",
       "description": "Queries (source|destination).(local_)ip",
       "label": "Query all IPs",
       "ext_type": "integer"
     }
   }
   ```
 * Events: Subqueries can need the parameter '%s' more or less than one time in the SQL query, for example:
   ```sql
   ("source.ip" = '%s' OR "destination.ip" = '%s')
   ```

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,7 +1,20 @@
 ## 0.5.2 to 0.5.3 (unreleased)
 
-(TODO)
-
+ * Events: Subqueries can be extended by the configuration, for example:
```
Comment (bernhardreiter) [inline NEWS.md]: The configuration example should go the example configuration file in server.py (if it is self explaining).

File: NEWS.md [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+   "subqueries": {
+     "all_ips": {
+       "sql": "(\"source.ip\" = %s OR \"source.local_ip\" = %s OR \"destination.ip\" = %s OR \"destination.local_ip\" = %s)",
+       "description": "Queries (source|destination).(local_)ip",
+       "label": "Query all IPs",
+       "ext_type": "integer"
+     }
+   }
+   ```
+ * Events: Subqueries can need the parameter '%s' more or less than one time in the SQL query, for example:
```
Comment (bernhardreiter) [inline NEWS.md]: Suggestion for improvement: I'm trying to write the NEWS.md entries from a user perspective, maybe this sentence can be improved. I had to read it a few times to understand it. What about:
```
 * Events: Enhances subqueries, so they can use an SQL query that uses a single parameter several times.
```

To have queries without a parameter, I think we'd would also have to change the frontend, wouldn't we?

**Comment by bernhardreiter** [comment]
Hi @wagner-certat,
thanks for the pull request. 
In general I think it is okay to make this configurable. 
I've left a few questions and suggestions in the diff.

Best Regards

File: NEWS.md (hunk 2 above)
Comment (wagner-certat) [inline NEWS.md]: > To have queries without a parameter, I think we'd would also have to change the frontend, wouldn't we?

A given parameter is just ignored

**Comment by wagner-certat** [comment]
I added the suggested changes, thanks for the feedback!

**Pull request merged**